### DISCUSSION TITLE
Daten an ThingSpeak (oder ähnlich) senden

### DISCUSSION THREAD

**Discussion by themanfrommoon** [tbnobody/OpenDTU#1134] [Q&A]
Moin,

ich vermute, dass man die Daten vom Wechselrichter über MQTT an (z.B.) ThingSpeak senden kann, und dann dort visualisieren kann.
Mein Ziel ist es eigentlich nur, ohne zusätzliche Hardware die Daten zu visualisieren. (Bitte nicht diskutieren, wieso, weshalb, warum. ICH selber habe eine Datenbank, und bei mir funktioniert das wunderbar. Ich suche eine Minimallösung für 2 Freunde, die keine Datenbank haben und auch nicht haben wollen.)

Ist das möglich?
Hat es schonmal jemand gemacht?

(Ich habe es gerade versucht, bin aber leider gescheitert. Ich habe es nicht hinbekommen, dass die MQTT Verbindung von openDTU zu ThingSpeak funktioniert...glaube ich jedenfalls.)

Beste Grüße,
Chris

**Comment by borti-16** [comment]
Hi,
ich schließ mich deiner Frage mal an. Findet er bei dir auch den Server aber kann sich nicht verbinden? (trotz 5mal geprüfter login daten...)
Disconnect reason:MQTT_MALFORMED_CREDENTIALS

Im thingspeak troubleshoot finde ich dass thingspeak auch auf "CleanSession flag 0" und "Will flag nonzero" mit CONNACK status of "0x05 Connection Refused, not authorized" reagiert. Allerdings kann ich in den Einstellungen oder auch im quellcode nirgends finden was hier bei openDTU gesetzt ist. Also ob es vielleicht einfach prinzipiell inkompatibel ist.

Hast du weiter probiert und ggf ne Lösung für dich gefunden? Meine Versuchen verlaufen bisher alle im Sand bzw. scheitern wahrscheinlich an meiner fehlenden Expertise ;)

Anmerkung: hab zum ersten mal mit MQTT zu tun und dachte thingspeak erlaubt mir danach mit Matlab weiter zu pfuschen (um mein Niveau damit auch korrekt zu beschreiben ;) ). Da ich ehrlich gesagt von Microcontroller-Programmierung keine Ahnung habe ist meine Hoffnung noch, dass ich irgendwas einfach nicht verstehe und es sich mit etwas hilfreicher Expertise doch einfach lösen lässt :)
Grüße
